### DISCUSSION TITLE
좋아요 기록 관련하여 의견

### DISCUSSION THREAD

**Discussion by JangGusWjd** [MajorIn-ABO/MajorIn_FE#7] [General]
현재는 좋아요 기록이 있을 때 한번 더 누르는 경우, 기록 자체를 아예 없애버리는 코드
삭제 시간을 저장하고 기록은 없애지 않는 것으로 바꾸는 쪽이 좋을지 고민 필요

**Comment by cho-ja-young** [comment]
### 스터디 게시글 테이블
- 유저id
….
- 좋아요 개수

### 스터디 좋아요 테이블
- 유저id
- 게시글id
…
- 좋아요

이렇게 테이블이 두개 있어서 좋아요 테이블에 누가 눌렀는지, 어떤 게시글에 눌렀는지 저장이 되고
여기서 일어나는 좋아요 누적이 위의 스터디 게시글 테이블의 좋아요 컬럼에 반영이 되는 구조

**Reply by JangGusWjd** [reply]
나는 기록 자체를 없애는게 좋을듯!
좋아요는 눌러서 좋아요 표시했다가 다시 눌러서 취소하고 또 다시 눌러서 좋아요 할 수 있는데, 삭제시간 저장하고 기록은 남아있으면 한 명의 사용자 당 기록이 너무 많아질 거 같아!

**Reply by cho-ja-young** [reply]
오키 지금은 일단
좋아요 취소시에 좋아요 테이블에서 좋아요가 아예 삭제되면서 
위의 스터디 게시글 테이블의 좋아요 컬럼 값이 하나 줄어들고 뭐 그런 구조인데

그러면 이 구조 그대로 가고 기존 게시글 좋아요도 이렇게 수정할게!!

**Reply by JangGusWjd** [reply]
오 좋아좋아~👍 지금 구조 그대로 가면 될듯! 게시글 좋아요도 같은 구조로!

**Reply by wlsl10** [reply]
웅웅 데이터 양은 최대한 줄이는 게 좋으니까 꼭 필요한 정보만 우선 넣어놓고 배포하고 나서 필요한 것들은 편리성을 위해 추가해도 될 것 같아
